apps/aecore/lib/aecore/chain/worker.ex

```diff
@@ -192,26 +203,26 @@ defmodule Aecore.Chain.Worker do
       "Added block ##{new_block.header.height} with hash #{Header.bech32_encode(new_block_hash)}, total tokens: #{inspect(total_tokens)}"
     end)
 
-    state_update1 = %{state | blocks_map: hundred_blocks_map,
+    state_update = %{state | blocks_map: hundred_blocks_map,
                               chain_states: updated_chain_states,
                               txs_index: new_txs_index}
+
     if top_height < new_block.header.height do
-      Persistence.batch_write(%{:chain_state => new_chain_state,
+      Persistence.batch_write(%{:chain_state => %{:chain_state => new_chain_state},
```

Comment:
Should be just %{chain_state: %{chain_state: new_chain_state}, and so on

Reply:
This is done for readability purpose